Assets/Source/IoT/Kinect/KinectManager.cs

```
@@ -29,42 +39,82 @@ public class KinectManager : IKinectManager
+            IMessageRouter messages,
+            MainCamera mainCamera)
         {
             _bootstrapper = bootstrapper;
+            _metrics = metrics;
             _messages = messages;
             _mainCamera = mainCamera;
         }
 
         /// <inheritdoc />
-        public IKinectContext Context()
+        public void SetIElementFactory(IElementFactory elementFactory)
+        {
+            _elementFactory = elementFactory;
+        }
+
+        /// <inheritdoc />
+        public IKinectCommonContext Context(KinectVersion version)
         {
-            return Context(null);
+            return Context(version, null);
         }
 
         /// <inheritdoc />
-        public IKinectContext Context(string kinectId)
+        public IKinectCommonContext Context(KinectVersion version, string kinectId)
         {
             // make sure it's non-null
             kinectId = kinectId ?? string.Empty;
 
-            var kinect = Windows.Kinect.KinectSensor.GetDefault();
-            if (null == kinect)
-            {
-                return new PassthroughKinectContext();
-            }
+            var configs = Object.FindObjectOfType<UnityPlatformConfigManager>();
+            var profile = configs.GetConfig().Profile;
+            var resources = configs.GetConfig().Resources;
 
-            if (!string.IsNullOrEmpty(kinectId) && kinect.UniqueKinectId != kinectId)
+            switch (version)
             {
-                return new PassthroughKinectContext();
-            }
+                case KinectVersion.Any:
+                {
+                    return _ctx ?? (_ctx = new KinectContext(_bootstrapper, _elementFactory, _metrics, _messages, _mainCamera, profile, resources));
+                }
+                case KinectVersion.AzureKinect:
+                {
+                    return _ctx ?? (_ctx = new KinectContext(_bootstrapper, _elementFactory, _metrics, _messages, _mainCamera, profile, resources));
```

I would think that if `_ctx` is not null and is not an azure kinect, it should throw an exception or log a warning.